parseedn.el

```diff
@@ -113,10 +113,24 @@ on available options."
         ((eq :lbracket token-type) (apply #'vector children))
         ((eq :set token-type) (list 'edn-set children))
         ((eq :lbrace token-type) (let* ((kvs (seq-partition children 2))
-                                        (hash-map (make-hash-table :test 'equal :size (length kvs))))
+                                        (hash-map (make-hash-table :test 'equal :size (length kvs)))
```

Comment:
Comparing the 'before' and 'after' of `defun parseedn-reduce-branch`, I sense that now it mixes levels of abstraction.

Before it looked more concise and higher-level.

The simplest fix would be to extract most of the logic to a separate defun. Although possibly this might mean something else (e.g. this could be fixed in parseclj?) - analysis much welcome.